### DISCUSSION TITLE
WPF Community-run projects

### DISCUSSION THREAD

**Discussion by gurpreet-wpf** [dotnet/wpf#6556] [Announcements]
### [!!!UPDATE!!!] 

We've realized that the term "community run projects" is a broad term and is open to different interpretations by developers. However, the intended meaning of this term was to have a channel through which we can prioritize issues/PRs that the community thinks would improve WPF over other existing open PRs/issues (https://github.com/orgs/dotnet/projects/146).

In future, we'll refer to such initiatives by a different name (TBD - To Be Decided) which may be less ambiguous.

Full description here: https://github.com/dotnet/wpf/discussions/7130#discussioncomment-4579285

### Original Post

Hello,

We are thrilled to announce that WPF is planning to start community-run projects which would be chosen and run by the community. This is in continuation to our efforts of increasing engagement with our community and to maximize and recognize your contributions which would help the overall community. 

We are planning to start first project for our upcoming release i.e. NET 7.0. Issues/Bugs (excluding feature requests/enhancements) will be chosen based on the upvotes/likes/reactions by the community and will be up for grabs for all our active contributors. We will try to close the PRs submitted for these issues on priority basis.

Start submitting the issues you want to be addressed by commenting them here below this post.

Scope of work items for this project:
	1. Issues
	2. Bugs

Not in scope:
	1. Feature requests
	2. Enhancements

Thanks!

**Comment by wstaelens** [comment]
XPS issues:
1. https://github.com/dotnet/wpf/issues/6301
2. https://github.com/dotnet/wpf/issues/3546

(3. https://github.com/dotnet/wpf/issues/4000 )
🤞

**Comment by lindexi** [comment]
- https://github.com/dotnet/wpf/issues/6415

**Comment by lindexi** [comment]
- https://github.com/dotnet/wpf/issues/3360

**Comment by lindexi** [comment]
- https://github.com/dotnet/wpf/issues/3145

**Comment by lindexi** [comment]
- https://github.com/dotnet/wpf/issues/2397

**Comment by lindexi** [comment]
- https://github.com/dotnet/wpf/issues/3147

**Comment by lindexi** [comment]
- https://github.com/dotnet/wpf/issues/6560

**Comment by olowk** [comment]
[XPS documents memory leak (System.Windows.ContextLayoutManager+LayoutQueue+Request and XpsDocument.GetFixedDocumentSequence())](https://github.com/dotnet/wpf/issues/6301)

**Comment by olowk** [comment]
[XPS documents in .NET Core can't be opened ](https://github.com/dotnet/wpf/issues/3546)

**Comment by JosefNemec** [comment]
[Laggy user experience when using VRR display](https://github.com/dotnet/wpf/issues/2294)

**Comment by cbaysinger** [comment]
Bugs:
Touches from dialog window get "stuck" in main window #450
Click event is rarely raised when using touch #194

**Comment by michael-hawker** [comment]
There's already been discussion in the .NET repo here: https://github.com/dotnet/runtime/issues/18087 with a community member engaged to work on it with a draft PR (https://github.com/dotnet/runtime/pull/65101).

They also want to fix it for WPF, but I believe need some support. Would be great to get some support and commitment from the WPF team to make this 6 year old request happen for .NET 7 (since it was originally pulled from .NET 5 https://github.com/dotnet/corefx/pull/35772).

WPF Issue: https://github.com/dotnet/wpf/issues/1887
WPF PR: https://github.com/dotnet/wpf/pull/6097

**Reply by Symbai** [reply]
First post says out of scope are feature requests and enhancement (which I personally find extremely disappointing). What you mention is unfortunately not a bug or issue so it would be out of scope. Like almost everything else people waiting for for a long time now.

**Reply by michael-hawker** [reply]
I would still argue it's a long-standing bug that the interface isn't respected. It's definitely a feature in .NET itself, but it's a bug in WPF that the interface isn't respected which would enable that feature.

Plus, there's already a community member up for helping on this one, so I think it fits the brief well.

**Comment by darcome** [comment]
Enhancement:
Create a control that can be used to embed 3d engines. Most of them create their own main window and cannot be embedded in wpf windows or other controls.

The control should be a surface that 3d engines would use to present their data.

The control should also be compatible with WinUI3 and MAUI.

**Comment by batzen** [comment]
Fixing TextBoxView memory leak for 2 seconds after unloading host control #1161

**Comment by batzen** [comment]
Select all text on triple click when no rich content #1163
=> I don't consider this a new feature but an accessibility bug fix

**Comment by batzen** [comment]
Use regular resource lookup for FocusVisualStyle #1165

**Comment by batzen** [comment]
Fixing validation adorner for not yet visible controls #1167

**Comment by batzen** [comment]
Preventing focus steal if parent window/application is not active #1169

**Comment by batzen** [comment]
Using a stopwatch and timespan instead of DateTime and uint #1171
=> This increases performance and improves reliability

**Reply by lindexi** [reply]
@batzen Try this format:

`- https://github.com/dotnet/wpf/pull/3286`

It will show as:

- https://github.com/dotnet/wpf/pull/3286

**Comment by batzen** [comment]
Replacing expensive type checks with "is" checks #4964

**Reply by IAmTheCShark** [reply]
My personal favourite.

**Comment by batzen** [comment]
Improve performance of DynamicResource usages #5610

**Comment by batzen** [comment]
Fix NRE when tooltip closing #2167

**Comment by batzen** [comment]
Less allocatey GetFullXmlns #3286

**Reply by bgrainger** [reply]
I don't think that PR is needed any longer: https://github.com/dotnet/wpf/pull/3286#issuecomment-1154432471

**Comment by batzen** [comment]
Remove the custom HashSet class #3738

**Comment by batzen** [comment]
Inline VerifyAccess #4021

**Comment by batzen** [comment]
Remove redundant static fields to reduce memory usage in XamlTypeInvoker #4027

**Comment by batzen** [comment]
Small performance improvement of PathParser #4208

**Comment by batzen** [comment]
CommandParameter invalidates CanExecute #4217

**Comment by batzen** [comment]
Merge pattern in DispatcherOperation #5296

**Comment by batzen** [comment]
Harden events against race conditions #5722

**Reply by lindexi** [reply]
This PR should be discussed.

**Comment by batzen** [comment]
I'd also really like to see the PRs from @ThomasGoulet73 regarding managed DWrite to be merged.

**Reply by Symbai** [reply]
We all do, but first post says bugs or issues only for whatever reason. So we will have to wait another year or five before that happens.

**Reply by batzen** [reply]
And your comment is for what exactly, except from being bitter?

**Comment by batzen** [comment]
Avoid excessive calls to the PropertyValues index getter. #6293

**Reply by wstaelens** [reply]
@batzen do you search these manually, by regex, ..? 

`new List[<].*[>][(\(\))?[ ]+[{]` regex to search for lists without an initial capacity.

`new[ ][a-zA-Z0-9]+\[0\]` regex to search for empty array init to replace with Array.Empty<T>

**Reply by batzen** [reply]
It's not my PR, i just suggested it's merge.

**Comment by batzen** [comment]
Use Dictionary instead of Hashtable in WeakEventTable #6443

**Comment by bgrainger** [comment]
- https://github.com/dotnet/wpf/pull/6632

**Comment by bgrainger** [comment]
- https://github.com/dotnet/wpf/pull/6254

**Comment by bgrainger** [comment]
- https://github.com/dotnet/wpf/pull/4231

**Comment by bgrainger** [comment]
- https://github.com/dotnet/wpf/pull/4966

**Comment by bgrainger** [comment]
- https://github.com/dotnet/wpf/pull/5218

**Comment by bgrainger** [comment]
- https://github.com/dotnet/wpf/pull/6252

**Comment by bgrainger** [comment]
- https://github.com/dotnet/wpf/pull/6272

**Comment by bgrainger** [comment]
- https://github.com/dotnet/wpf/pull/6280

**Comment by bgrainger** [comment]
- https://github.com/dotnet/wpf/pull/6285

**Comment by bgrainger** [comment]
- https://github.com/dotnet/wpf/pull/6500

**Comment by bgrainger** [comment]
- https://github.com/dotnet/wpf/pull/6501

**Comment by bgrainger** [comment]
- https://github.com/dotnet/wpf/pull/6502

**Comment by bgrainger** [comment]
- https://github.com/dotnet/wpf/pull/6511

**Comment by gurpreet-wpf** [comment]
Thanks everyone for submitting their responses. We are triaging the issues and PRs mentioned here. In order to kickstart the next steps, we would be closing this discussion now. However, if you still want to submit any more issues, feel free to comment here by 22nd June 2022. Thanks!

**Reply by wstaelens** [reply]
really hope for the XPS issues/improvements. 10k+ customers would for sure benefit from this. 🤞

**Comment by gurpreet-wpf** [comment]
Hi All,

We have triaged all the responses we have received here and based on the feasibility, we have selected a total of 24 PRs which would be tracked as part of this project -> https://github.com/orgs/dotnet/projects/146

We are targeting to close majority of these PRs within next 3 CTPs. [Sep 22]

Thanks
